### DISCUSSION TITLE
Implement watcher pattern for externalName services

### DISCUSSION THREAD

**Discussion by brianehlert** [nginx/kubernetes-ingress#4816] [Ideas]
Support dynamic updating of externalName services.
Currently NGINX Ingress Controller primarily relies on the NGINX Plus capability of the dynamic upstream resolver to continuously resolve DNS names for upstreams (NGINX is always routing to IP addresses under the hood).

NGINX OSS relies on a different DNS resolution path that is tied to start and reload and does not dynamically query and seamlessly update externalName services.

This is different from Service resolution that uses the K8s API to resolve pod IP addresses using endpointSlices for upstreams.

While both of these achieve the task of updating the upstream servers when DNS names are provided for externalName Services the gap is that the project does not subscribe a watcher to the externalName Service resource.  And thus if the resource is changed or deleted the corresponding configurations are not updated immediately.  The references to the externalName Services must be removed / updated from the VirtualServer, TransportServer, or Ingress resources.

If an externalName service is configured in an Ingress, VirtualServer, TransportServer what needs to happen?
- should the corresponding configuraiton be removed from NGINX.conf?
   - This would mean that an externalName reference might be set in in a VS/Ingress and the configuration not applied
-  should this create an error condition for the Ingress/VS that references the externalName Service?

**Comment by brianehlert** [comment]
@shaun-nx Was this solved with the additional work implemented for upstream:backup?

**Comment by jjngx** [comment]
it's already implemented
